src/import/pull-job.c

```diff
@@ -73,6 +75,31 @@ static void pull_job_finish(PullJob *j, int ret) {
                 j->on_finished(j);
 }
 
+static int pull_job_restart(PullJob *j) {
+        int r;
+        char *chksum_url = NULL;
+
+        r = import_url_change_last_component(j->url, "SHA256SUMS", &chksum_url);
+        if (r < 0)
+                return r;
+
+        j->url = chksum_url;
+        j->state = PULL_JOB_INIT;
+        j->payload = NULL;
+        j->payload_size = 0;
+        j->payload_allocated = 0;
+        j->written_compressed = 0;
+        j->written_uncompressed = 0;
+        j->written_since_last_grow = 0;
```

Comment:
maybe I am missing something, but don't you have to free the old `j->url` and `j->payload` here first?